packages/expo-updates/android/src/main/java/expo/modules/updates/db/dao/AssetDao.java

```diff
@@ -124,4 +136,14 @@ public boolean addExistingAssetToUpdate(UpdateEntity update, AssetEntity asset,
 
     return deletedAssets;
   }
+
+  @Transaction
+  public void markMissingAssets(List<AssetEntity> missingAssets) {
+    ArrayList<Long> missingAssetIds = new ArrayList<>();
+    for (AssetEntity asset : missingAssets) {
+      missingAssetIds.add(asset.id);
+    }
+    _markMissingAssets(missingAssetIds);
+    _markUpdatesWithMissingAssets(missingAssetIds, UpdateStatus.PENDING);
```

Comment:
This seems like it should be in the same transaction.

Reply:
(since these functions are one off, one way could be to combine them into a single SQL call: https://stackoverflow.com/questions/2044467/how-to-update-two-tables-in-one-statement-in-sql-server-2005/2044520#2044520)

Reply:
That's actually exactly what the `@Transaction` annotation does here, since I believe Room's `@Query` helper can only include a single statement.

Reply:
oh i see! awesome.